src/django_components/slots.py

```diff
@@ -303,114 +328,119 @@ def _try_parse_as_named_fill_tag_set(
         elif isinstance(node, TextNode) and node.s.isspace():
             pass
         else:
-            return None
+            return []
     return result
 
 
 def _try_parse_as_default_fill(
     nodelist: NodeList,
     ComponentNodeCls: Type[Node],
-) -> Optional[ImplicitFillNode]:
+) -> Sequence[FillNode]:
     nodes_stack: List[Node] = list(nodelist)
     while nodes_stack:
         node = nodes_stack.pop()
-        if isinstance(node, NamedFillNode):
-            return None
+        if isinstance(node, FillNode):
+            return []
         elif isinstance(node, ComponentNodeCls):
             # Stop searching here, as fill tags are permitted inside component blocks
             # embedded within a default fill node.
             continue
         for nodelist_attr_name in node.child_nodelists:
             nodes_stack.extend(getattr(node, nodelist_attr_name, []))
     else:
-        return ImplicitFillNode(nodelist=nodelist)
-
-
-def _block_has_content(nodelist: NodeList) -> bool:
```

Comment:
Moved `_block_has_content` to `node.py`, since there is now a couple of helper functions all related to working with `Node` or `Nodelists`